### ISSUE TITLE
Pasting from clipboard instead of cutbuffer?

### ISSUE THREAD

**Issue by Unknow0059** [lhmouse/nano-win#15]
Is this supported, or am I missing something?

### Expected:
- Copy something outside nano
- Paste with ^U or ^V or ^Insert
- Copied text gets pasted

###  Occurred:
- Copy something outside nano
- Paste with ^U or ^V or ^Insert
- Error: "Cutbuffer is empty"

**Comment by lhmouse** [comment]
No, it is not supported. Nano has no access to the Windows clipboard. Normally it is the terminal driver that does that, which means inside CMD you may only select text and copy it by right-clicking. Note CMD itself does not support non-rectangle selection.

Not only is this the case on Windows, it is also what people do on Linux. See <https://www.nano-editor.org/dist/v5/faq.html#4.4>.

**Comment by Unknow0059** [comment]
Ok.

What about code highlighting? [As described here](https://askubuntu.com/questions/90013/how-do-i-enable-syntax-highlighting-in-nano).

**Comment by lhmouse** [comment]
It is supported, but the bundled `.nanorc` contains Unix paths which you have to replace with Windows paths such as `D:/nano/syntax/*.nanorc`.

**Comment by Unknow0059** [comment]
Where exactly should the bundled `.nanorc` be placed in order for nano to recognize it?
In the archive, it is placed above the directory that comes with nano's bin and share folders. Doesn't seem to be recognized there. Nor on %USERPROFILE% for that matter.
Also, if .nanorc is supported, how come `set` isn't? If I try `set wordwrap` I get a `Not supported on Windows` error.

**Comment by lhmouse** [comment]
`.nanorc.` should be placed in your home directory. On my computer it is `C:/Users/lhmouse`. You should ensure it is named `.nanorc`, not `.nanorc.nanorc` or `.nanorc.text` whatsoever, as explorer does not display extensions by default.

Where did you get the `Not supported on Windows` error? It only happens when you attempt to suspend nano (with **Ctrl-Z**) or to execute a command (with **Ctrl-T**, or **Ctrl-R** followed by **Ctrl-X**).

**Comment by Unknow0059** [comment]
As previously mentioned, `.nanorc` is not recognized in %USERPROFILE%, which is an environment variable that on my computer expands to `C:/Users/` and then my Windows account's name.

Error:

![nano_unsupported_bug](https://user-images.githubusercontent.com/25656892/96668706-2251a280-1332-11eb-8d5f-6df69d8f908b.gif)

**Comment by lhmouse** [comment]
I suppose you run nano with admin privilege? Then `%ALLUSERSPROFILE%/.nanorc` is used instead. See https://github.com/lhmouse/nano-win/commit/4e3ff45b6046b17937166d1b98355de6828307ac#diff-3e04e056edbdc0f772b0ffd1c93cef1187209f37276d40d2a281d9b989398a73R41.

**Comment by Unknow0059** [comment]
Placing `.nanorc` in %ALLUSERSPROFILE% doesn't seem to have had any effect.
Of course, I've made sure that some settings from there have been uncommented.
Specifically the ones from under:
`## Paint the interface elements of nano.  These are examples;`
Which would be easily viewed when in effect.

**Comment by lhmouse** [comment]
I have no idea. Perhaps you can build nano locally, and use GDB to set breakpoints on the `have_nanorc()` and `parse_one_nanorc()` functions, to find out whether the global variable `nanorc` is assigned a valid path.

You cannot use the _execute command_ feature to turn on/off options. To enable/disable soft wrapping. press **Alt-$** i.e. **Alt-Shift-4** on an American keyboard.

**Comment by Unknow0059** [comment]
This week I had been hibernating my computer instead of shutting it down. Yesterday I was having some issues with `explorer.exe`  and Virtual Desktops so I finally restarted my computer.
Today I tried running nano-win again with `.nanorc` in %USERPROFILE% and the settings applied as expected. It seems the lack of shutdown had been causing issues here, too.

I have the environment variable %HOME% set to a folder within that directory in the hopes that I can keep all my Unix files in one place instead of scattered in Users. 
For that reason it is inconvenient that nano-win ignores the variable, but I am very glad that it's just working now.
